libs/payments/errors.go

```diff
@@ -0,0 +1,40 @@
+package payments
+
+import (
+	"fmt"
+)
+
+// PaymentError is an error used to communicate whether an error is temporary.
+type PaymentError struct {
+	OriginalError  error
+	FailureMessage string
+	Temporary      bool
+}
+
+// Error makes ProcessingError an error
+func (e PaymentError) Error() string {
+	msg := fmt.Sprintf("error: %s", e.FailureMessage)
+	if e.Cause() != nil {
```

Comment:
issue: when we set both the `OriginalError` and `FailureMessage` and call `Error()` we end up printing the error message twice.

```
err := errors.New("raw message")
p := ProcessingErrorFromError(err, true)

p.Error() // error: raw message: raw message
```
The `if` condition also suggest we expect a `PaymentError` to have a `FailureMessage` but no `OriginalError` which is not in keeping with `ProcessingErrorFromError` and a little unclear.

suggestion: we simplify the type to just contain an single error messages or derive the `FailureMessage` from the stored `OriginalError` and remove the FailureMessage.

We could also store a slice of error message 

```
type PaymentError struct {
    errors []string
    temporary bool
}
```

Reply:
I've simplified the type, but recommend that we wait to finalize its structure until we integrate a payout method and have a real case to consider.

Reply:
Additionally, for most of the types here, you want receivers be pointers, not values.